oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/ElasticsearchClientFactory.java

```diff
@@ -61,7 +64,6 @@ public RestHighLevelClient getConnection(ElasticsearchCoordinate esCoord) {
     public void close() {
         lock.writeLock().lock();
         try {
-            isClosed.set(true);
```

Comment:
Usually it's a good idea to try avoiding to close twice... why did you remove it?

Reply:
I had to remove it because the `getConnection` could not be used anymore when closed (see the throw Exception I removed). This was a problem in both the tests and also when the OSGI bundle gets deactivated and then re-activated.
Looking again at the code I figured it out the concurrent map needs to be cleared on close to avoid returning already closed clients: I have changed the code.

Reply:
Here we are keeping all clients in a map and in a multi threaded environment if a thread call close we will be closing all and clearing complete map. Shouldn't we close connections opened by the specific thread only?